mesh/mesh.go

```diff
@@ -551,61 +560,48 @@ func (msh *Mesh) AddBallot(
 	ctx context.Context,
 	ballot *types.Ballot,
 ) (*wire.MalfeasanceProof, error) {
+START:
+	initiallyMalfeasant := false
+	msh.addBallotMu.Lock()
 	malicious := msh.atxsdata.IsMalicious(ballot.SmesherID)
 	if malicious {
 		ballot.SetMalicious()
+		initiallyMalfeasant = true
+	}
+
+	// Store should only allow one ballot per node ID per batch.
+	waitC, errFunc, retry := msh.ballotWriter.Store(ballot)
+	msh.addBallotMu.Unlock()
```

Comment:
Does `msh.atxsdata.IsMalicious()` must be in the locked section? How about:
```suggestion
	malicious := msh.atxsdata.IsMalicious(ballot.SmesherID)
	if malicious {
		ballot.SetMalicious()
		initiallyMalfeasant = true
	}

	// Store should only allow one ballot per node ID per batch.
	msh.addBallotMu.Lock()
	waitC, errFunc, retry := msh.ballotWriter.Store(ballot)
	msh.addBallotMu.Unlock()
```

Reply:
The motivation behind putting it inside the lock was the following:
- on the writer side (here) we want to check whether the identity is malicious and set the ballot as malicious before adding it, under lock, because:
- on the consumer side (in the forever loop `Start`) we hold the lock during the whole execution time of writing the batch, and during that time, an identity might turn into a malicious one


This is also why I eventually went away from doing a CoW for the new batch (allowing to queue writes while another batch is already being written) among other reasons (also because of the complexity of keeping tabs over the single-ballot-per-identity-per-batch part).
So TL;DR this is done this way because of at least a hypothetical data race I was trying to avoid. That being said, an identity can _still_ turn into a malicious one for other reasons during that time due to other protocols.
I'm not sure if this makes too much sense though, let me know your thoughts on this. Happy to make the proposed change.

Reply:
Ok actually given this some thought and the mutex here might not be at all needed because of the locking strategy inside the ballot writer.

Reply:
I ended up removing the mutex